src/plugins/analysis/file_system_metadata/docker/mount.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+import pickle
+from contextlib import contextmanager
+from pathlib import Path
+from subprocess import CalledProcessError, check_call
+
+INPUT_DIR = Path("/work")
+INPUT_FILE = INPUT_DIR / "input.img"
+MOUNT_DIR = Path("/root/mount_dir")
+OUTPUT_FILE = INPUT_DIR / "output.pickle"
+
+
+@contextmanager
+def mount(input_file: Path):
+    try:
+        check_call(f'mount -o ro,loop {input_file} {MOUNT_DIR}', shell=True)
+        yield
+    finally:
+        check_call(f'umount {MOUNT_DIR}', shell=True)
+
+
+def main():
+    try:
+        stats = INPUT_FILE.lstat()
+        with mount(INPUT_FILE):
+            result = _get_mounted_file_stats()
+        _save_results(result, stats.st_uid, stats.st_gid)
+    except FileNotFoundError:
+        logging.error('Could not find the input file.')
+    except CalledProcessError:
+        logging.error('Error during mount.')
+
+
+def _get_mounted_file_stats():
+    result = []
+    for file in MOUNT_DIR.rglob("*"):
+        try:
+            if not file.is_symlink() and file.is_file():
+                result.append((file.name, str(file.relative_to(MOUNT_DIR)), file.lstat()))
+        except PermissionError:
+            continue
+    return result
```

Comment:
There does not appear to be a particular reason to use pickle here instead of json (other than us having to convert `Path.lstat` to a dict. We should try to avoid pickle as much as possible.

Reply:
changed